pkg/reconciliation/reconcile_racks.go

```diff
@@ -2092,16 +2094,152 @@ func (rc *ReconciliationContext) CheckConditionInitializedAndReady() result.Reco
-		if err == nil {
-			break
+		if podJobId, found := pod.Annotations[podJobIdAnnotation]; found {
+			if features.Supports(httphelper.AsyncSSTableTasks) {
+				_, found := pod.Annotations[podJobStatusAnnotation]
+				if !found {
+					// Pod is currently processing something, or has finished processing.. update the status
+					details, err := rc.NodeMgmtClient.JobDetails(pod, podJobId)
+					if err != nil {
+						rc.ReqLogger.Error(err, "Could not get JobDetails for pod", "Pod", pod)
+						return result.Error(err)
+					}
+
+					if details.Id == "" {
+						// This job was not found, pod most likely restarted. Let's retry..
+						delete(pod.Annotations, podJobIdAnnotation)
+						err = rc.Client.Update(context.Background(), pod)
+						if err != nil {
+							return result.Error(err)
+						}
+						return result.RequeueSoon(2)
+					} else if details.Status == podJobError {
+						// Log the error, move on
+						rc.ReqLogger.Error(fmt.Errorf("cleanup failed: %s", details.Error), "Pod", pod)
+						pod.Annotations[podJobStatusAnnotation] = podJobError
+						err = rc.Client.Update(context.Background(), pod)
+						if err != nil {
+							return result.Error(err)
+						}
+						continue
+					} else if details.Status == podJobCompleted {
+						// Pod has finished, remove the job_id and let us move to the next pod
+						pod.Annotations[podJobStatusAnnotation] = podJobCompleted
+						err = rc.Client.Update(context.Background(), pod)
+						if err != nil {
+							return result.Error(err)
+						}
+						continue
+					} else if details.Status == podJobWaiting {
+						// Job is still running or waiting
+						return result.RequeueSoon(2)
+					}
+				} else {
+					// This pod has finished since it has a status
+					continue
+				}
+			} else {
+				if len(jobRunner) > 0 {
+					// Something is still holding the worker
+					return result.RequeueSoon(2)
+				}
+
+				// Nothing is holding the job, has this pod finished or hasn't it ran anything?
+				if _, found := pod.Annotations[podJobStatusAnnotation]; found {
+					// Pod has finished, check next one
+					continue
+				}
+			}
 		}
+
+		if features.Supports(httphelper.AsyncSSTableTasks) {
+			// Pod isn't running anything at the moment, this pod should run next
+			jobId, err := rc.NodeMgmtClient.CallKeyspaceCleanup(pod, -1, "", nil)
+			if err != nil {
+				// We can retry this later, it will only restart the cleanup but won't otherwise hurt
+				return result.Error(err)
+			}
+			pod.Annotations[podJobHandler] = jobHandlerMgmtApi
+			pod.Annotations[podJobIdAnnotation] = jobId
+
+			err = rc.Client.Update(context.Background(), pod)
+			if err != nil {
+				rc.ReqLogger.Error(err, "Pod", pod)
+				return result.Error(err)
+			}
+		} else {
+			jobId := strconv.Itoa(idx)
+
+			// This pod should run next, mark it
+			pod.Annotations[podJobHandler] = oplabels.ManagedByLabelValue
+			pod.Annotations[podJobIdAnnotation] = jobId
+
+			err = rc.Client.Update(context.Background(), pod)
```

Comment:
Would patch be better here? We don't need to worry about conflicts do we? 

Shouldn't be using `context.Background()`.